From fc62aeceea92169efb5a336511a55275f605ec02 Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Mon, 8 Sep 2014 12:26:03 -0400
Subject: [ethernet] tg3: remove open-coded skb_cow_head

Message-id: <1410179173-14304-10-git-send-email-ivecera@redhat.com>
Patchwork-id: 91667
O-Subject: [RHEL7.1 PATCH 09/19] tg3: remove open-coded skb_cow_head.
Bugzilla: 1088637
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

BZ: #1088637

Upstream commit(s):
commit 105dcb5979e70fee7a4dbd6505ec89d406258f1a
Author: francois romieu <romieu@fr.zoreil.com>
Date:   Sat Mar 29 12:26:29 2014 +0100

    tg3: remove open-coded skb_cow_head.

    Signed-off-by: Francois Romieu <romieu@fr.zoreil.com>
    Cc: Nithin Nayak Sujir <nsujir@broadcom.com>
    Cc: Michael Chan <mchan@broadcom.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/broadcom/tg3.c b/drivers/net/ethernet/broadcom/tg3.c
index 150580d..b8a1621 100644
--- a/drivers/net/ethernet/broadcom/tg3.c
+++ b/drivers/net/ethernet/broadcom/tg3.c
@@ -7917,8 +7917,7 @@ static netdev_tx_t tg3_start_xmit(struct sk_buff *skb, struct net_device *dev)
   struct iphdr *iph;
   u32 tcp_opt_len, hdr_len;
 
-  if (skb_header_cloned(skb) &&
-      pskb_expand_head(skb, 0, 0, GFP_ATOMIC))
+  if (skb_cow_head(skb, 0))
    goto drop;
 
   iph = ip_hdr(skb);
-- 
1.7.1